people-and-planet-ai/land-cover-classification/e2e_test.py

```diff
@@ -0,0 +1,442 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import csv
+import importlib
+import io
+import logging
+import os
+import platform
+import subprocess
+import sys
+from unittest.mock import Mock, patch
+import uuid
+
+import apache_beam as beam
+from apache_beam.io.filesystems import FileSystems
+from apache_beam.testing.test_pipeline import TestPipeline
+import ee
+import google.auth
+from google.cloud import aiplatform
+from google.cloud import storage
+import nbconvert
+import nbformat
+import numpy as np
+import pytest
+import requests
+import tensorflow as tf
+
+import batch_predict
+import create_datasets
+import trainer
+
+
+PYTHON_VERSION = "".join(platform.python_version_tuple()[0:2])
+
+NAME = f"ppai/land-cover-py{PYTHON_VERSION}"
+
+UUID = uuid.uuid4().hex[0:6]
+PROJECT = os.environ["GOOGLE_CLOUD_PROJECT"]
+LOCATION = "us-central1"
+
+IPYNB_FILE = "README.ipynb"
+PY_FILE = "README.py"
+
+PATCH_SIZE = 16
+INPUTS_DTYPE = [(name, "<f8") for name in trainer.INPUT_BANDS]
+NUM_CLASSES = trainer.NUM_CLASSIFICATIONS
+
+# Colab libraries are not available, so we disable them explicitly.
+sys.modules["google.colab"] = Mock()
+
+
+# Add more logging information for debugging.
+logging.getLogger().setLevel(logging.INFO)
+
+
+@pytest.fixture(scope="session")
+def bucket_name() -> str:
+    storage_client = storage.Client()
+
+    bucket_name = f"{NAME.replace('/', '-')}-{UUID}"
+    bucket = storage_client.create_bucket(bucket_name, location=LOCATION)
+
+    logging.info(f"bucket_name: {bucket_name}")
+    yield bucket_name
+
+    subprocess.check_call(["gsutil", "-m", "rm", "-rf", f"gs://{bucket_name}/*"])
+    bucket.delete(force=True)
+
+
+@pytest.fixture(scope="session")
+def container_image() -> str:
+    # https://cloud.google.com/sdk/gcloud/reference/builds/submit
+    container_image = f"gcr.io/{PROJECT}/{NAME}:{UUID}"
+    subprocess.check_call(
```

Comment:
There's also a Python SDK if you want to use that instead.

Reply:
I can see how this would be simpler though.

Reply:
I would prefer to leave it like this for simplicity, it also matches more closely what is done in the notebook.